### ISSUE TITLE
Unhandled reply jumps out of branch into other stories

### ISSUE THREAD

**Issue by xavieramoros** [wit-ai/node-wit#132]
When you have multiple stories, any unexpected user input makes the bot jump to random stories.

Let me explain it looking at the official SlotBased example (https://wit.ai/l5t/SlotBased_Example/stories/202b6764-4b70-4bc6-8406-437138f9b368): 

In the missingTime branch, if the user replies with any message, findTheater is still executed as expected after the jump:
![screen shot 2017-02-02 at 15 40 58](https://cloud.githubusercontent.com/assets/347030/22553728/0cb6394c-e95e-11e6-93bd-e507d3d8b1d3.png)

![screen shot 2017-02-02 at 15 34 59](https://cloud.githubusercontent.com/assets/347030/22553502/39c9f884-e95d-11e6-90fe-8b6e938b7f16.png)
 
On the contrary, in my own bot, in a very similar scenario instead of going back to a bot action after a jump, the conversation jumps to a random story.

This is my bot story:
![screen shot 2017-02-02 at 15 37 21](https://cloud.githubusercontent.com/assets/347030/22553596/a081b0ee-e95d-11e6-80af-6c8eaa2dbeaf.png)

This is my testing the bot:
![screen shot 2017-02-02 at 15 37 54](https://cloud.githubusercontent.com/assets/347030/22553598/a4076fce-e95d-11e6-9077-c2510010647b.png)

After Whatever it should call again my function getVenueTimed but it's not the case. Something weird is happening.

**Comment by kyle-ssg** [comment]
+1, I saw things quickly fall apart as soon as I started implementing multiple stories

**Comment by nemphys** [comment]
Right, whenever I tried "jumping" to a bookmark like this, it never worked. Had to revert to using the same method call as a new step (instead of the "jump").
